### PR TITLE
Support building with GHC 8.2

### PR DESCRIPTION
**Pull request by 23Skidoo**
`AllowAmbiguousTypes` requires GHC 7.8, if we want to support older versions we'll have to use `CPP`.

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Added CPP to support older GHCs.

**Comment by 23Skidoo** [comment]
@Deewiant Travis is green, please consider merging and releasing a new version on Hackage.

**Comment by Deewiant** [comment]
Sorry, I've been rather ill lately. I'll try to take a look towards the end of the week.

**Comment by 23Skidoo** [comment]
Get well!

**Comment by Deewiant** [comment]
Ok, finally managed to get around to this. Some overall comments:

* `.gitignore` is a good point, this project pre-dates me understanding that those should always come included. As a pet peeve I'd rather not have highly workflow-specific things in there though so no `TAGS`, please.
* Adding `other-extensions` is also a good point, thanks.
* `cabal.project` is also fine.
* Using `AllowAmbiguousTypes` is quite a sledgehammer in this case, especially due to the requirement to use `CPP` to enable it only when supported. The offending function can be fixed directly by providing a tighter type signature or simplifying `let c = differenceWith j a b` to `let c = go a b`.
* Requiring `cabal-install` 1.24 everywhere is unacceptable. As the comments in the `.travis.yml` indicate, the first two lines map exactly to what the corresponding Debian distro provides out of the box (e.g. for Debian 8 it is 1.20: https://packages.debian.org/jessie/cabal-install), and I want to support that directly. I suppose that for now I'll just conditionally `--allow-newer` based on `$GHCVER`.

In conclusion, thanks for spurring me into action but sorry that I can't accept your PR as-is. I'll cherry-pick your `other-extensions` and `cabal.project` changes but otherwise sort things out myself and push out a new release once I can get Travis happy with everything.

**Comment by Deewiant** [comment]
https://hackage.haskell.org/package/list-tries-0.6.4

**Comment by 23Skidoo** [comment]
Great, thanks! FWIW, `--allow-newer` is only supported since 1.20.
